diff --git a/crates/arti/src/proxy.rs b/crates/arti/src/proxy.rs
index 07ca143e9..99d63e2d4 100644
--- a/crates/arti/src/proxy.rs
+++ b/crates/arti/src/proxy.rs
@@ -34,6 +34,7 @@ use anyhow::{Context, Result, anyhow};
 
 /// Placeholder type when RPC is disabled at compile time.
 #[cfg(not(feature = "rpc"))]
+#[cfg_attr(feature = "experimental-api", non_exhaustive)]
 #[cfg_attr(feature = "experimental-api", visibility::make(pub))]
 pub(crate) enum RpcMgr {}
 
diff --git a/crates/arti/src/rpc_stub.rs b/crates/arti/src/rpc_stub.rs
index 27dbe193b..0cb3d7554 100644
--- a/crates/arti/src/rpc_stub.rs
+++ b/crates/arti/src/rpc_stub.rs
@@ -3,5 +3,4 @@
 /// An uninhabited type, for use when constructing a proxy listener.
 ///
 /// (This type _is_ inhabited when we're built with RPC support.)
-#[cfg_attr(feature = "experimental-api", visibility::make(pub))]
 pub(crate) enum RpcProxySupport {}
diff --git a/crates/fs-mistrust/src/file_access.rs b/crates/fs-mistrust/src/file_access.rs
index 1b60abf9d..07c56f149 100644
--- a/crates/fs-mistrust/src/file_access.rs
+++ b/crates/fs-mistrust/src/file_access.rs
@@ -277,6 +277,7 @@ impl<'a> FileAccess<'a> {
             .write_all(contents.as_ref())
             .map_err(|e| Error::io(e, &tmp_name, "write to file"))?;
         // Flush and close.
+        #[cfg_attr(target_arch = "wasm32", expect(clippy::drop_non_drop))]
         drop(tmp_file);
 
         // Replace the old file.
diff --git a/crates/hashx/src/register.rs b/crates/hashx/src/register.rs
index 021cec081..51848414d 100644
--- a/crates/hashx/src/register.rs
+++ b/crates/hashx/src/register.rs
@@ -36,7 +36,7 @@ impl RegisterId {
     /// (Recall that hashx has 8 virtual registers,
     /// so the output of this method is always in range 0..=7.)
     #[inline(always)]
-    #[cfg(feature = "compiler")]
+    #[cfg(all(feature = "compiler", not(target_arch = "wasm32")))]
     pub(crate) fn as_u8(&self) -> u8 {
         self.0
     }
diff --git a/crates/tor-config-path/src/addr.rs b/crates/tor-config-path/src/addr.rs
index 97fb20f8d..de8e61837 100644
--- a/crates/tor-config-path/src/addr.rs
+++ b/crates/tor-config-path/src/addr.rs
@@ -56,7 +56,7 @@ impl CfgAddr {
                 #[cfg(not(unix))]
                 {
                     // Give this error early on non-unix platforms, so that we don't confuse the user.
-                    return Err(unix::NoAfUnixSocketSupport::default().into());
+                    Err(unix::NoAfUnixSocketSupport::default().into())
                 }
                 #[cfg(unix)]
                 {
diff --git a/crates/tor-dirserver/Cargo.toml b/crates/tor-dirserver/Cargo.toml
index 87ef8e2d0..83a151266 100644
--- a/crates/tor-dirserver/Cargo.toml
+++ b/crates/tor-dirserver/Cargo.toml
@@ -43,7 +43,6 @@ rand = "0.9.2"
 retry-error = { version = "0.11.0", path = "../retry-error" }
 # NOTE: See the note for `rusqlite` in `../tor-dirmgr/Cargo.toml`.
 rusqlite = ">=0.36.0, <0.39.0"
-saturating-time = "0.3.0"
 strum = { version = "0.28.0", features = ["derive"] }
 thiserror = "2.0.16"
 tokio = { version = "1.47.1", features = ["full", "test-util"] }
diff --git a/crates/tor-netdoc/Cargo.toml b/crates/tor-netdoc/Cargo.toml
index c74d6930a..40402b9b1 100644
--- a/crates/tor-netdoc/Cargo.toml
+++ b/crates/tor-netdoc/Cargo.toml
@@ -114,7 +114,6 @@ memchr = "2.5"
 paste = "1.0.3"
 phf = { version = "0.13.1", features = ["macros"] }
 rand = { version = "0.9.1", optional = true }
-saturating-time = "0.3.0"
 serde = { version = "1.0.103", features = ["derive"] }
 serde_with = "3.0.0"
 signature = "2"
diff --git a/crates/tor-ptmgr/src/ipc.rs b/crates/tor-ptmgr/src/ipc.rs
index 62fb5d1e6..5ddb870b9 100644
--- a/crates/tor-ptmgr/src/ipc.rs
+++ b/crates/tor-ptmgr/src/ipc.rs
@@ -435,6 +435,7 @@ pub(crate) mod sealed {
                 // Dropping stdin should tell the PT to exit, since we set the correct environment
                 // variable for that to happen.
                 trace!("Asking PT {} to exit, nicely.", ident);
+                #[cfg_attr(target_arch = "wasm32", expect(clippy::drop_non_drop))]
                 drop(_stdin);
                 // Give it some time to exit.
                 thread::sleep(GRACEFUL_EXIT_TIME);
