diff --git a/crates/tor-chanmgr/src/builder.rs b/crates/tor-chanmgr/src/builder.rs
index fbc761770..3a181e79f 100644
--- a/crates/tor-chanmgr/src/builder.rs
+++ b/crates/tor-chanmgr/src/builder.rs
@@ -145,8 +145,7 @@ where
 #[async_trait]
 impl<R: Runtime, H: TransportImplHelper> ChannelFactory for ChanBuilder<R, H>
 where
-    R: tor_rtcompat::TlsProvider<H::Stream> + Send + Sync,
-    H: Send + Sync,
+    R: tor_rtcompat::TlsProvider<H::Stream>,
 {
     #[instrument(skip_all, level = "trace")]
     async fn connect_via_transport(
@@ -176,8 +175,7 @@ where
 #[async_trait]
 impl<R: Runtime, H: TransportImplHelper> IncomingChannelFactory for ChanBuilder<R, H>
 where
-    R: tor_rtcompat::TlsProvider<H::Stream> + Send + Sync,
-    H: Send + Sync,
+    R: tor_rtcompat::TlsProvider<H::Stream>,
 {
     type Stream = H::Stream;
 
diff --git a/crates/tor-chanmgr/src/transport.rs b/crates/tor-chanmgr/src/transport.rs
index 27ab1b7ce..933aa4324 100644
--- a/crates/tor-chanmgr/src/transport.rs
+++ b/crates/tor-chanmgr/src/transport.rs
@@ -34,7 +34,7 @@ use tor_rtcompat::StreamOps;
 // TODO: Maybe move this to a separate crate so that tor-ptmgr can be
 // used without having to depend on chanmgr.
 #[async_trait]
-pub trait TransportImplHelper {
+pub trait TransportImplHelper: Send + Sync {
     /// The type of the resulting stream.
     type Stream: AsyncRead + AsyncWrite + StreamOps + Send + Sync + 'static;
 
diff --git a/crates/tor-chanmgr/src/transport/proxied.rs b/crates/tor-chanmgr/src/transport/proxied.rs
index 64bbb7d14..3f995a090 100644
--- a/crates/tor-chanmgr/src/transport/proxied.rs
+++ b/crates/tor-chanmgr/src/transport/proxied.rs
@@ -437,7 +437,7 @@ impl<R: NetStreamProvider + Send + Sync> ExternalProxyPlugin<R> {
 
 #[cfg(feature = "pt-client")]
 #[async_trait]
-impl<R: NetStreamProvider + Send + Sync> TransportImplHelper for ExternalProxyPlugin<R> {
+impl<R: NetStreamProvider> TransportImplHelper for ExternalProxyPlugin<R> {
     type Stream = R::Stream;
 
     async fn connect(&self, target: &OwnedChanTarget) -> crate::Result<(PeerAddr, R::Stream)> {
